src/main/java/sparksoniq/spark/closures/LetClauseMapClosure.java

```diff
@@ -17,33 +17,61 @@
  * Author: Stefan Irimescu
  *
  */
- package sparksoniq.spark.closures;
+package sparksoniq.spark.closures;
 
-import org.apache.spark.api.java.function.Function;
+import org.apache.spark.api.java.function.MapFunction;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.types.StructType;
 import sparksoniq.jsoniq.item.Item;
 import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
-import sparksoniq.jsoniq.tuple.FlworTuple;
 import sparksoniq.semantics.DynamicContext;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class LetClauseMapClosure implements Function<FlworTuple, FlworTuple> {
-    private final String _variableName;
+public class LetClauseMapClosure implements MapFunction<Row, Row> {
     private final RuntimeIterator _expression;
+    StructType _oldSchema;
 
-    public LetClauseMapClosure( String variableName, RuntimeIterator expression) {
+    public LetClauseMapClosure(RuntimeIterator expression, StructType oldSchema) {
         this._expression = expression;
-        this._variableName = variableName;
+        this._oldSchema = oldSchema;
     }
+
     @Override
-    public FlworTuple call(FlworTuple v1) throws Exception {
-        List<Item> result = new ArrayList<>();
-        _expression.open(new DynamicContext(v1));
-        while (_expression.hasNext())
-            result.add(_expression.next());
+    public Row call(Row row) throws Exception {
+        String[] columnNames = _oldSchema.fieldNames();
+
+        // Deserialize row
+        List<Object> deserializedRow = ClosureUtils.deserializeEntireRow(row);
+        List<List<Item>> rowColumns = new ArrayList<>();
+        for (Object columnObject : deserializedRow) {
+            List<Item> column = (List<Item>) columnObject;
+            rowColumns.add(column);
+        }
+
+        // Create dynamic context with deserialized data
+        DynamicContext context = new DynamicContext();
+        for (int columnIndex = 0; columnIndex < columnNames.length; columnIndex++) {
+            context.addVariableValue(columnNames[columnIndex], rowColumns.get(columnIndex));
+        }
+
+        // Apply expression to the context
+        List<Item> newColumn = new ArrayList<>();
+        _expression.open(context);
+        while (_expression.hasNext()) {
+            Item nextItem = _expression.next();
+            newColumn.add(nextItem);
+        }
         _expression.close();
-        v1.putValue(_variableName, result, true);
-        return v1;
+
+        List<byte[]> newRowColumns = new ArrayList<>();
+        for (List<Item> column : rowColumns) {
+            newRowColumns.add(ClosureUtils.serializeItemList(column));
```

Comment:
Can we not just copy over the existing binary values from the former row? Serializing again looks redundant and is a performance overhead. I think only the new column needs to be serialized.

This comment may also apply to the for closure if the same is done here, and that could help improve performance.